src/chat/fetchingReducer.js

```diff
@@ -51,7 +52,8 @@ export default (state: FetchingState = initialState, action: Action): FetchingSt
       return messageFetchStart(state, action);
 
     case MESSAGE_FETCH_COMPLETE:
-      return messageFetchComplete(state, action);
+      // We don't want to accumulate old searches that we'll never need again.
+      return isSearchNarrow(action.narrow) ? state : messageFetchComplete(state, action);
```

Comment:
nit: Let's add the new logic around `isSearchNarrow` (and the comment that goes with it) inside the `messageFetchComplete` handler function, instead of here.

Similarly for `messageFetchStart` (when adding to that per my other comment about keeping them "parallel"), and in `narrowsReducer`.

In general, the way we structure all these reducers is:
* If the code for a given case is simple enough that it feels like it fits inline in the `switch` in the reducer function, we put it there.
* Otherwise we have one of these handler functions named after the action type, and that case in the handler just calls the handler which does all the work.

That's helpful because either way it keeps all the logic in one place -- if you see a function with a name like `messageFetchComplete` in a reducer file, you know it completely describes what we do in that reducer for a `MESSAGE_FETCH_COMPLETE` action.

Reply:
Done.